core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -4525,8 +4525,12 @@ object Stream extends StreamLowPriority {
       * res3: scodec.bits.ByteVector = ByteVector(5 bytes, 0x0001020304)
       * }}}
       */
-    def to(collector: Collector[O]): G[collector.Out] =
-      compiler(underlying, () => collector.newBuilder)((acc, c) => { acc += c; acc }, _.result)
+    def to(collector: Collector[O]): G[collector.Out] = {
+      implicit val G: Monad[G] = compiler.target
+      G.unit.map(_ => collector.newBuilder).flatMap { builder =>
```

Comment:
It's ok since it's very localised, but this appears a third time (here, compile token, and somewhere else), it probably benefits from a `suspendSynchronousInfallible` or something, with a `Compiler.Target` bound and the same scaladoc that's in `Token`

Reply:
BTW, note that we don't have a `Compiler.Target[G]` here -- I originally tried making the new `compiler.target` val return a `Compiler.Target[G]` but we don't have an inductive instance for `Compiler.Target[F] => Compiler.Target[Resource[F, *]]`

Reply:
ah, I missed this. Well, nevermind then :)